build_tools/riscv/riscv_bootstrap.sh

```diff
@@ -11,110 +11,77 @@
+  if [[ -d "${BOOTSTRAP_WORK_DIR}" ]]; then
+    rm -rf "${BOOTSTRAP_WORK_DIR}"
   fi
 }
 
 # Call the cleanup function when this tool exits.
 trap cleanup EXIT
 
-wget_google_drive() {
-  local file_id="$1"
-  local file_name="$2"
-  local install_path="$3"
-  local tar_option="$4"
-
-  wget --save-cookies ${BOOTSTRAP_WORK_DIR}/cookies.txt \
-    "https://docs.google.com/uc?export=download&id="$file_id -O- | \
-    sed -En "s/.*confirm=([0-9A-Za-z_]+).*/\1/p" > ${BOOTSTRAP_WORK_DIR}/confirm.txt
-  wget --progress=bar:force:noscroll --load-cookies ${BOOTSTRAP_WORK_DIR}/cookies.txt \
-    "https://docs.google.com/uc?export=download&id=$file_id&confirm=`cat ${BOOTSTRAP_WORK_DIR}/confirm.txt`" -O- | \
-    tar $tar_option - --no-same-owner --strip-components=1 -C $install_path
-}
-
+# Download and install the toolchain from IREE-OSS GCS
 download_file() {
-  # server name or google drive file_id
-  local file_download_info="$1"
-  local file_name="$2"
-  local install_path="$3"
-  # download method(e.g. wget_google_drive)
-  local download_method="$4"
-
-  echo "Install $2 to $3"
-  if [[ -e $3/file_info.txt ]]; then
+  local file_name="$1"
+  local install_path="$2"
+  local file_sha="$3"
+
+  echo "Install $1 to $2"
+  if [[ "$(ls -A $2)" ]]; then
     read -p "The file already exists. Keep it (y/n)? " replaced
     case ${replaced:0:1} in
       y|Y )
-        echo "Skip download $2."
+        echo "Skip download $1."
         return
       ;;
       * )
-        rm -rf $3
+        rm -rf $2
       ;;
     esac
   fi
 
-  local tar_option=""
-  if [[ "${file_name##*.}" == "gz" ]]; then
-    tar_option="zxpf"
-  elif [[ "${file_name##*.}" == "bz2" ]]; then
-    tar_option="jxpf"
-  fi
-  echo "tar option: $tar_option"
-
-  echo "Download $file_name ..."
+  echo "Download ${file_name} ..."
   mkdir -p $install_path
-  $download_method $file_download_info $file_name $install_path $tar_option
-
-  echo "$file_download_info $file_name" > $install_path/file_info.txt
+  wget --progress=bar:force:noscroll -P "${BOOTSTRAP_WORK_DIR}" "${IREE_ARTIFACT_URL}/${file_name}" && \
+    echo "${file_sha} ${BOOTSTRAP_WORK_DIR}/${file_name}" | sha256sum -c -
+  echo "Extract ${file_name} ..."
+  tar -C "${install_path}" -xf "${BOOTSTRAP_WORK_DIR}/${file_name}" --no-same-owner --strip-components=1
 }
 
-mkdir -p ${BOOTSTRAP_WORK_DIR}
+mkdir -p "${BOOTSTRAP_WORK_DIR}"
 
 read -p "Install RISCV clang toolchain(y/n)? " answer
 case ${answer:0:1} in
   y|Y )
-    download_file ${RISCV_CLANG_TOOLCHAIN_FILE_ID} \
-                  ${RISCV_CLANG_TOOLCHAIN_FILE_NAME} \
-                  ${TOOLCHAIN_PATH_PREFIX} \
-                  wget_google_drive
+    download_file "${RISCV_CLANG_TOOLCHAIN_FILE_NAME}" \
+                  "${TOOLCHAIN_PATH_PREFIX}" \
+                  "${RISCV_CLANG_TOOLCHAIN_FILE_SHA}"
+
     echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
     echo " PLEASE run 'export RISCV_TOOLCHAIN_ROOT=${TOOLCHAIN_PATH_PREFIX}'   "
```

Comment:
A trick you can use to enable you to set these things yourself is to have the user `source` the script instead of running it. You can even detect whether the user correctly sourced it and error out if they ran it instead: https://stackoverflow.com/q/2683279

Reply:
Thanks! However, using source here messed with trap and bash autocomplete (causing the shell session to be terminated when I try to use tab to autocomplete the CLI later). I can look into it later, but keep the direct bash script call for now.